myapp/app/models/task.rb

```diff
@@ -2,4 +2,11 @@
 
 class Task < ApplicationRecord
   validates :title, presence: true, length: { maximum: 256 }
+
+  def self.sort_task_by(params)
```

Comment:
controllerからparamをまるっともらうよりは必要な情報のみをもらう形の方がいいかなと思います。
今はindexからしか使われませんが、今後他のactionから使おうとした場合に呼び元のactionによってこちらに渡される配列の構造が異なるとなるとtestとかしづらそうです。
```suggestion
  def self.sort_task_by(sort, direction)
```